### ISSUE TITLE
Collaboration: Intune protected App with Microsoft Apps (Storage being considered as Personal)

### ISSUE THREAD

**Issue by eg-aseem** [msintuneappsdk/ms-intune-app-sdk-ios#345]
Hello,

We have an app that allows browsing files on a cloud and displays a preview of these files (i.e. Office files). Additionally, we allow users to add our storage location within Office apps.

As discussed in https://github.com/msintuneappsdk/ms-intune-app-sdk-ios/issues/325, it's not possible to directly share a file with Office app and edit it so we are looking for a way to make editing Office documents as easy as possible.

Users are able to "Share" a document (via native share options) to i.e. Microsoft Word, but the document is opened in read-only mode. There is an option to "Save a copy", but if we select it and try to save a file in our storage, we get a message "Your organisation doesn't allow saving in personal locations". Is there any way we can make this scenario work? Anything we can do, so the storage is not considered "personal"?

Thank you very much in advance for any help.

**Comment by eg-aseem** [comment]
Any thoughts or pointers on this issue!

**Comment by Kyle-Reis** [comment]
Hi @eg-aseem. Users' ability to save to your cloud storage solution would be dependent on the outgoing data transfer and save-as policy that the IT admin has configured for the Office application in the Intune admin portal:

![image](https://github.com/msintuneappsdk/ms-intune-app-sdk-ios/assets/7443106/2dc88544-2bdf-4191-9905-dacdd1e3ab1c)

**Comment by eg-aseem** [comment]
Hey @Kyle-Reis , thank you very much for responding. We are aware of these settings in Intune admin portal. From what we have learned so far, in order to be able to save Office documents in our cloud storage (without lowering securiting settings of the policy), it needs to be: a) *not* considered "Personal" in Office365 apps (this is the case for OneDrive, see: screenshot) b) enabled on the list under "Allow user to save copies to selected services" (from your screenshot). The trouble is - we don't know how to achieve that. Is there some kind of Microsoft certification process that we have to initiate on our side? Perhaps there is some documentations about this topic? We would greatly appreciate any hints.

To sum up, we would like to keep the policy as you have shown:
* "Send org data to other apps" set at least to "Policy managed apps"
* "Save copies of org data" set to blocked, but with our service enabled, but currently it is not listed there, comparing to i.e. OneDrive for Business or Box.

<img src="https://github.com/msintuneappsdk/ms-intune-app-sdk-ios/assets/129847029/cb30e649-243d-4c1a-8034-30963fb89107" data-canonical-src="https://github.com/msintuneappsdk/ms-intune-app-sdk-ios/assets/129847029/cb30e649-243d-4c1a-8034-30963fb89107" width="400" />

**Comment by Kyle-Reis** [comment]
Hi @eg-aseem, I think the first step here would be to apply for the [Cloud Storage Partner Program](https://developer.microsoft.com/office/cloud-storage-partner-program) and then submit a request to join the [Cloud Storage Partner Program Plus](https://learn.microsoft.com/microsoft-365/cloud-storage-partner-program/plus/overview) if you meet those qualifications.

**Comment by Kyle-Reis** [comment]
Closing out this issue as the original question has been answered. Please reach out again if you have any other issues or questions.
